include/eve/module/core/regular/impl/simd/x86/roundscale.hpp

```diff
@@ -16,189 +16,50 @@
-    requires(is_one_of<D>(types<toward_zero_type, downward_type, to_nearest_type, upward_type> {})
-             && x86_abi<abi_t<T, N>>)
-{
-  return D()(roundscale[true_(as(a0))])(a0, s);
-}
-
-// -----------------------------------------------------------------------------------------------
-// Masked case
-template<int S, conditional_expr C, floating_scalar_value T, typename N>
-EVE_FORCEINLINE wide<T, N>
-                roundscale_(EVE_SUPPORTS(avx512_),
-                            C const                                      &cx,
-                            wide<T, N> const                             &a0,
-                            std::integral_constant<int, S> s) noexcept requires x86_abi<abi_t<T, N>>
-{
-  return to_nearest(roundscale[cx])(a0, s);
-}
-
-template<int R, typename WT> struct rm_imm_choice
-{
-  rm_imm_choice() {}
-
-  template<typename F>
-  EVE_FORCEINLINE auto operator()(WT src, auto m, WT a, int s, F f) const noexcept
+  template<auto S, floating_scalar_value T, typename N, callable_options O>
+  EVE_FORCEINLINE wide<T, N>roundscale_(EVE_SUPPORTS(avx512_),
+                                        O const          &o,
+                                        wide<T, N> const &a0,
+                                        std::integral_constant<int, S> const & ) noexcept
+  requires x86_abi<abi_t<T, N>>
   {
-    switch( (s) )
-    {
-    case 0: return f(src, m, a, std::integral_constant<int, (0 << 4) + R>()); break;
-    case 1: return f(src, m, a, std::integral_constant<int, (1 << 4) + R>()); break;
-    case 2: return f(src, m, a, std::integral_constant<int, (2 << 4) + R>()); break;
-    case 3: return f(src, m, a, std::integral_constant<int, (3 << 4) + R>()); break;
-    case 4: return f(src, m, a, std::integral_constant<int, (4 << 4) + R>()); break;
-    case 5: return f(src, m, a, std::integral_constant<int, (5 << 4) + R>()); break;
-    case 6: return f(src, m, a, std::integral_constant<int, (6 << 4) + R>()); break;
-    case 7: return f(src, m, a, std::integral_constant<int, (7 << 4) + R>()); break;
-    case 8: return f(src, m, a, std::integral_constant<int, (8 << 4) + R>()); break;
-    case 9: return f(src, m, a, std::integral_constant<int, (9 << 4) + R>()); break;
-    case 10: return f(src, m, a, std::integral_constant<int, (10 << 4) + R>()); break;
-    case 11: return f(src, m, a, std::integral_constant<int, (11 << 4) + R>()); break;
-    case 12: return f(src, m, a, std::integral_constant<int, (12 << 4) + R>()); break;
-    case 13: return f(src, m, a, std::integral_constant<int, (13 << 4) + R>()); break;
-    case 14: return f(src, m, a, std::integral_constant<int, (14 << 4) + R>()); break;
-    case 15: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>()); break;
-    default: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>());
-    }
+    constexpr auto c = categorize<wide<T, N>>();
+    constexpr auto s = ((S) << 4) + rounding_mode<O>(toward_zero2);
```

Comment:
Again, the defautl value was to_nearest, why the change ?

Reply:
an error
corrected